commit fcab594a3030d49db933d68bdad398ab8fea265c
Author: Michał Winiarski <michal.winiarski@intel.com>
Date:   Mon Jul 6 16:41:07 2020 +0200

    drm/i915: Don't taint when using fault injection
    
    It is not really unexpected to hit wedge on init this way.
    We're already downgrading error printk when running with fault injection,
    let's use the same approach for CI tainting.
    
    v2: Don't check fault inject in trace dump (Chris)
    
    Signed-off-by: Michał Winiarski <michal.winiarski@intel.com>
    Cc: Chris Wilson <chris@chris-wilson.co.uk>
    Cc: Michal Wajdeczko <michal.wajdeczko@intel.com>
    Cc: Petri Latvala <petri.latvala@intel.com>
    Reviewed-by: Chris Wilson <chris@chris-wilson.co.uk>
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
    Link: https://patchwork.freedesktop.org/patch/msgid/20200706144107.204821-3-michal@hardline.pl

diff --git a/drivers/gpu/drm/i915/i915_utils.c b/drivers/gpu/drm/i915/i915_utils.c
index 01a3d3c941bf..4c305d838016 100644
--- a/drivers/gpu/drm/i915/i915_utils.c
+++ b/drivers/gpu/drm/i915/i915_utils.c
@@ -53,7 +53,10 @@ void add_taint_for_CI(struct drm_i915_private *i915, unsigned int taint)
 {
 	__i915_printk(i915, KERN_NOTICE, "CI tainted:%#x by %pS\n",
 		      taint, (void *)_RET_IP_);
-	__add_taint_for_CI(taint);
+
+	/* Failures that occur during fault injection testing are expected */
+	if (!i915_error_injected())
+		__add_taint_for_CI(taint);
 }
 
 #if IS_ENABLED(CONFIG_DRM_I915_DEBUG)